Codereview/bortoletti

bortoletti-giorgia opened the pull request:
## Premise

Hi, for the app of the group1 I took care of the management of contacts.
Our app was designed to ask for a location or to ring a phone.
The phone number can be entered manually or selected from an address book.
Accordingly I created an address book where user can add, edit or delete a contact.
The edit is made by swiping to the right. Instead the deletion is made with a swipe left on the contact. This is associated with an undo operation.
A search on the name and number can also be done.
Once a contact is selected, its number is shown on the home page.

## Changes

- `SMSContactManager `to manage contacts in the database
- `Adapter` to manage the RecyclerView with contacts list
- `SwipeCallback` to manage the swipe to right and left
- `ContactFilter` to filter contacts in the list
- `SearchListener` to start filter for contacts

## Side Changes

I have attached some files to be reviewed in full and others that you can look at for completeness.
Full revision is required when there is _all the code_. If you want you can review the rest of code (marked with `//CODE REVIEW`) but it is not the priority
If you want you can download the branch and test it with the emulator.

### Footer

This is one of the first times that I do interaction between graphics and backend so I accept advice very gladly. Thank you.

bortoletti-giorgia committed:
fix: add classes for code review

bortoletti-giorgia committed:
style: modify some specs

bortoletti-giorgia committed:
fix: modify some specs

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSwipeCallback.java  5
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java  2
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java  6
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5093766&bid=16754738)

bortoletti-giorgia committed:
fix: mark tests for code review

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java  2
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSwipeCallback.java  5
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java  3
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java  6
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5093766&bid=16754738)

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java:
```diff
@@ -0,0 +1,78 @@
+     * @return {@link android.widget.Filter.FilterResults} results after filtering
+     */
+    @Override
+    protected FilterResults performFiltering(CharSequence constraint) {
```
Consider to always add annotations to method signatures (and more in general to the code), they make it more readable, intuitive and, sometimes, they save you some checks. In this case, the value `null` for the parameter `constraint` is admitted (meaning that there is nothing to filter) so the annotation to put before the parameter is `@Nullable`.

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java:
```diff
@@ -0,0 +1,78 @@
+                }
+                if (item.getAddress().contains(filterPattern)) {
+                    filteredList.add(item);
+                }
```
In my opinion these two statements can be compressed into one with an OR operator. Functionalities won't be affected but you will save one if statement and an instruction duplication.

bortoletti-giorgia replied:
Absolutely yes, I corrected.

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/ContactManager.java:
```diff
@@ -0,0 +1,93 @@
+    /**
+     * Return all contacts present in the database
+     *
+     * @return an array of {@link C} saved in the database
```
To be more precise the returned value is a list.
```suggestion
     * @return a {@link List} of {@link C} saved in the database
```

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/ContactRecyclerHelper.java:
```diff
@@ -0,0 +1,28 @@
+
+    /**
+     * Return {@link C} to the given position.
+     *
```
Here you missed the `@param` tag for `position`.

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java:
```diff
@@ -0,0 +1,181 @@
+     */
+    private SMSContactManager(@NonNull Context applicationContext) throws ExceptionInInitializerError {
+        if(instance != null)
+            throw new ExceptionInInitializerError();
```
This check is, in this case, not actually needed because the constructor call is only done when the instance is `null` and therefore the check will always return false. If you are not planning to call the constructor elsewhere in this class (by creating new `SMSContactManager` objects, that shouldn't be done according to the Singleton Design Pattern) consider removing this check and the consequent exception throwing.

bortoletti-giorgia replied:
This check was inserted precisely to remedy the fact that the constructor was invoked within the class itself. It shouldn't be necessary. Now I think if I remove it, thank you very much.

matteocarnelos replied:
As I said, if you are not planning to call the constructor anymore (and you should) it can be removed.

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java:
```diff
@@ -0,0 +1,78 @@
+    protected FilterResults performFiltering(CharSequence constraint) {
+        List<C> filteredList = new ArrayList<>();
+
+        if (constraint == null || constraint.length() == 0) {
+            filteredList.addAll(allContacts);
+        } else {
```
Small personal styling suggestion: I like to put single-instruction if statements all in one line, or, if they are too long, in two lines without brackets because, by doing this, the code become more compact in my opinion. You are totally free to choose what you prefer, it's not an issue.

bortoletti-giorgia replied:
```suggestion
        if (constraint == null || constraint.length() == 0) filteredList.addAll(allContacts);
        else {
```

Like this?

matteocarnelos replied:
Exactly, that's personal taste so if you don't like it you can leave it as it was. Remember that if you decide to change the convention you should modify all the code accordingly.

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java:
```diff
@@ -0,0 +1,78 @@
+    private SMSContactRecyclerAdapter adapter;
+
+    /**
+     * Constructor
```
As I've seen through the code, specifications are not stylistically perfectly consistent (overall they are, but there are some small discrepancies, like the presence of the ending dot, the starting upper case character,...) and sometimes, like in this case, not very explanatory. Consider to even them and to add, also when there isn't much to say, a small explanation of what the method does. This will increase the readability and beauty of the code.

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java:
```diff
@@ -0,0 +1,181 @@
+     */
+    public void addContact(@NonNull SMSPeer peer) {
+        SMSContact newContact = SMSContactConverterUtils.contactFromPeer(peer);
+        contactDatabase.access().insert(newContact);
```
Here you can use method nesting taking advantage of the last method overload, that is `addContact(@NonNull SMSContact newContact)`, by doing this:
```suggestion
        addContact(newContact);
```
Right now there isn't any big advantage (not even a line saved), but if one day, in the future, the process of adding a new contact to the database become more complex then you will have to modify only the last overload instead of all the overloads.

bortoletti-giorgia replied:
You are right.
In other places I used it but here I forgot.

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java:
```diff
@@ -0,0 +1,181 @@
+     */
+    public void addContact(@NonNull SMSPeer peer, @NonNull String nameContact) {
+        SMSContact newContact = SMSContactConverterUtils.contactFromPeer(peer, nameContact);
+        contactDatabase.access().insert(newContact);
```
Same here, you can use method nesting:
```suggestion
        addContact(newContact);
```

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java:
```diff
@@ -0,0 +1,181 @@
+     */
+    public void removeContact(@NonNull SMSPeer peer) {
+        SMSContact oldContact = SMSContactConverterUtils.contactFromPeer(peer);
+        contactDatabase.access().delete(oldContact);
```
Same thing here too, advantages are not evident now but otherwise it is a possible source of bugs in the future.
```suggestion
        removeContact(oldContact);
```

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java:
```diff
@@ -0,0 +1,244 @@
+     * to represent an item.
+     *
+     * @param parent
+     * @param viewType
```
Here the parameter description is missing. Consider adding a brief explanation.

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java:
```diff
@@ -0,0 +1,244 @@
+                        notifyItemChanged(selectedPosition);
+                        selectedPosition = DEFAULT_SELECTED_POSITION;
+                    }
+                    MainActivity.setContactSelected(getSelectedItem());
```
I tested the app and I found the contact selection system a little slow, in order to select a contact you have to open your custom contact picker, select one and manually close the activity. I would have personally preferred a behavior more similar to the system contact picker, which it automatically closes itself when a contact is selected.
Practically you have to notify here the `ContactListActivity` that a contact has been selected and it can close itself.
By changing the behavior you don't need the yellow highlighting anymore and you can convert the `ContactListActivity` to an activity for result, avoiding the `MainActivity.setContactSelected(...)` method.

If you like to leave it like it is now no problem, this is just a suggestion.

bortoletti-giorgia replied:
Done, thank you.
Further changes will be made in the next days because Riccardo has transformed these Activities into Fragments and therefore I will update the methods in Fragment with more precise specifications.

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java:
```diff
@@ -0,0 +1,43 @@
+     *
+     * @param query     Text to search
+     *
+     * @return false, as default
```
Please consider adding a brief explanation on what the value `false` does, because it is not clear.

bortoletti-giorgia replied:
This method is invoked by default when the user adds o removes a letter from the search bar. I don't call it somewhere. Therefore I don't save or view the value that returns. I wrote about 'default' because nothing determines true or false. 
I don't know how explain it. Do you have any suggestion?

matteocarnelos replied:
When you are not sure about descriptions in overridden methods try to look at the parent method, as it will probably contain well detailed descriptions. For example, if you look at `onQueryTextSubmit(...)` you will find this info:
```java
/**
* Called when the user submits the query. This could be due to a key press on the
* keyboard or due to pressing a submit button.
* The listener can override the standard behavior by returning true
* to indicate that it has handled the submit request. Otherwise return false to
* let the SearchView handle the submission by launching any associated intent.
*
* @param query the query text that is to be submitted
*
* @return true if the query has been handled by the listener, false to let the
* SearchView perform the default action.
*/
boolean onQueryTextSubmit(String query);
```
So you can modify your specification accordingly:
```suggestion
     * @return false, letting the SearchView perform the default action.
```
The same thing can be done also in the method below.

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java:
```diff
@@ -0,0 +1,43 @@
+     *
+     * @param newText   Text to search
+     *
+     * @return false, as default
```
Same here, consider adding an explanation on the `false` returned value.

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSwipeCallback.java:
```diff
@@ -0,0 +1,139 @@
+     *
+     * @param recyclerView View that contains items
+     * @param viewHolder   Item to move
+     * @param target       Its position is the final position for viewHolder
```
Here you missed the `@return` tag.

matteocarnelos reviewed (changes requested):
### Code Readability
The code is overall well formatted and readable, I've left some suggestions to improve the compactness but you are free to follow them or not. Variables name have been chosen correctly and naming conventions are good and kept consistent. There are only some small formatting discrepancies in method specifications that should be fixed, you can refer to my comments.
### Structure and functioning
The code is overall very well structured, however some changes can be made in order to make it more robust and safe from bugs, for example the addition of annotations in all the methods (see my comments). Functionalities have been applied well and they seem to work fine. The only thing that I wasn't so happy about was the base concept of working, in my opinion some changes can be done in order to make the user interaction more fluid and fast, I've left my explanation about this with a comment at the end of `SMSContactRecyclerAdapter`.
### Documentation and testing
Method specifications are present in every method and they are informative most of the time, however there are some missed tags and descriptions, please consider to fix this. For what I've seen tests cover most of the code.
### Personal notes
Honestly I'm a bit doubtful about the choice of creating your own contact picker instead of using the system one because the user can't select contacts that he has added to che contact list of his phone but instead he has to recreate and adding them into your contact database. Anyway, except this doubt the realization of this custom contact picker is very well made and I liked it a lot, the choice of saving everything into a database is right in my opinion and I particularly liked the filtering system. 

Good job!

bortoletti-giorgia committed:
style: modify some specs and code

After Carnelos's review.

bortoletti-giorgia committed:
fix: ContactListActivity is an activity for result

Now when user has clicked on an contact, the activity closes automatically and the phone number is found in the home.

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 11
           

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSwipeCallback.java  5
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java  3
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java  6
- app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5093766&bid=16754738)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactConverterUtils.java:
```diff
@@ -1,5 +1,7 @@
 package ingsw.group1.findmyphone.contacts;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'androidx.annotation.Nullable'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5093766)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused import - androidx.annotation.Nullable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5093766)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java:
```diff
@@ -0,0 +1,226 @@
+import androidx.core.content.ContextCompat;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.security.AccessController;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'java.security.AccessController'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5093766)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused import - java.security.AccessController.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5093766)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java:
```diff
@@ -0,0 +1,226 @@
+
+import androidx.annotation.NonNull;
+import androidx.constraintlayout.widget.ConstraintLayout;
+import androidx.core.content.ContextCompat;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused import - androidx.core.content.ContextCompat.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5093766)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'androidx.core.content.ContextCompat'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5093766)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java:
```diff
@@ -0,0 +1,226 @@
+import ingsw.group1.findmyphone.R;
+import ingsw.group1.findmyphone.activity.ActivityConstantsUtils;
+import ingsw.group1.findmyphone.activity.ContactListActivity;
+import ingsw.group1.findmyphone.activity.MainActivity;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'ingsw.group1.findmyphone.activity.MainActivity'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5093766)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused import - ingsw.group1.findmyphone.activity.MainActivity.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5093766)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java:
```diff
@@ -0,0 +1,226 @@
+
+import android.app.Activity;
+import android.content.Intent;
+import android.graphics.Color;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused import - android.graphics.Color.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5093766)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'android.graphics.Color'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5093766)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java:
```diff
@@ -93,42 +91,20 @@ public void onClick(View v) {
-            txtPhoneNumber.setText("");
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if(resultCode == RESULT_OK && requestCode == ActivityConstantsUtils.REQUEST_SELECTION_CONTACT)
+            if(data.hasExtra(ActivityConstantsUtils.SELECTED_PHONE_NUMBER))
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5093766)

bortoletti-giorgia committed:
style: remove unused import

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java  3
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java  6
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSwipeCallback.java  5
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5093766&bid=16754738)

bortoletti-giorgia committed:
fix: remove exception in private constructor of Singleton SMSContactManager

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java  3
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java  6
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSwipeCallback.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5093766&bid=16754738)

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java:
```diff
@@ -0,0 +1,221 @@
+
+                    data.putExtra(ActivityConstantsUtils.SELECTED_PHONE_NUMBER, selectedContact.getAddress());
+                    activity.setResult(RESULT_OK, data);
+                    activity.onBackPressed(); //back to home
```
Here you can use the `finish()` method, it is more appropriate.
```suggestion
                    activity.finish(); //back to home
```

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java:
```diff
@@ -0,0 +1,41 @@
+public class SMSContactSearchListener implements SearchView.OnQueryTextListener {
+
+    private SMSContactRecyclerAdapter recyclerAdapter;
+
```
I noticed now that here specifics are missing.

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSwipeCallback.java:
```diff
@@ -0,0 +1,139 @@
+    private SMSContactRecyclerAdapter contactAdapter;
+
+    /**
+     * Constructor
```
Add some explanations here.

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java:
```diff
@@ -0,0 +1,221 @@
+        public ConstraintLayout layout;
+
+        /**
+         * Constructor
```
Add some explanations here.

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java:
```diff
@@ -0,0 +1,221 @@
+    //---------------------------- CONSTRUCTOR ----------------------------
+
+    /**
+     * Constructor
```
Add some explanations here.

matteocarnelos commented on app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java:
```diff
@@ -0,0 +1,80 @@
+     * @param results    {@link android.widget.Filter.FilterResults} contacts to show after filtering
+     */
+    @Override
+    protected void publishResults(@Nullable CharSequence constraint, FilterResults results) {
```
Annotation missing on `results` parameter and also in other methods through the code. I'm not reporting all of them but consider completing the annotations.

matteocarnelos reviewed (approved):
Wow! You have applied all of my suggestions, that's what I was talking about! Very good job!
I'm very happy about the conversion of `ContactListActivity` in activity for result, I'm glad that you followed my advice.
However it's not perfect yet, there are still some changes to do, like adding annotations, specifics and descriptions (see my comments), but it's everything about code readability and not about functioning, so I'll approve these changes anyway now. I also left a comment about what method to use in order to exit from an activity.

Approved!🎉

bortoletti-giorgia committed:
style: add specs, all dots and @NonNull annotations

bortoletti-giorgia commented:
Thank you so much for your review.
Excellent advice, especially for the activity for result. I added all the points and @NonNull annotations.

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java  6
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSwipeCallback.java  5
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5093766&bid=16754738)
